packages/expo-router/src/link/useLinkToPathProps.tsx

```diff
@@ -52,9 +73,23 @@ export default function useLinkToPathProps({ href, ...options }: UseLinkToPathPr
     }
   };
 
+  const baseAppendedStrippedHref = React.useMemo(() => {
+    const strippedHref = stripGroupSegmentsFromPath(href);
+    // Ensure there's always a value for href.
+    if (!strippedHref) {
+      return appendBaseUrl('/');
+    }
+
+    // Append base url only if needed.
+    if (shouldAppendBaseUrl(strippedHref)) {
```

Comment:
Use `shouldLinkExternally` https://github.com/expo/expo/blob/df05000d67f74aae9c5e03a5d9c5b95807220025/packages/expo-router/src/utils/url.ts#L16

Reply:
Changed! Didn't spot that one existing already.